jib-gradle-plugin/src/test/java/com/google/cloud/tools/jib/gradle/GradleProjectPropertiesTest.java

```diff
@@ -420,63 +390,27 @@ public void testCreateContainerBuilder_war()
             "/my/app/WEB-INF/classes/package/Other.class");
   }
 
-  @Test
-  public void testCreateContainerBuilder_defaultWebAppRoot()
-      throws URISyntaxException, IOException, InvalidImageReferenceException,
-          CacheDirectoryCreationException {
-    Path unzipTarget = setUpWarProject(getResource("gradle/webapp"));
-
-    BuildContext buildContext = setupBuildContext(JavaContainerBuilder.DEFAULT_WEB_APP_ROOT);
-    ContainerBuilderLayers layers = new ContainerBuilderLayers(buildContext);
-    assertThat(layers.dependenciesLayer.getEntries())
-        .comparingElementsUsing(SOURCE_FILE_OF)
-        .containsExactly(unzipTarget.resolve("WEB-INF/lib/dependency-1.0.0.jar"));
-    assertThat(layers.snapshotsLayer.getEntries())
-        .comparingElementsUsing(SOURCE_FILE_OF)
-        .containsExactly(unzipTarget.resolve("WEB-INF/lib/dependencyX-1.0.0-SNAPSHOT.jar"));
-    assertThat(layers.resourcesLayer.getEntries())
-        .comparingElementsUsing(SOURCE_FILE_OF)
-        .containsExactly(
-            unzipTarget.resolve("META-INF"),
-            unzipTarget.resolve("META-INF/context.xml"),
-            unzipTarget.resolve("Test.jsp"),
-            unzipTarget.resolve("WEB-INF"),
-            unzipTarget.resolve("WEB-INF/classes"),
-            unzipTarget.resolve("WEB-INF/classes/empty_dir"),
-            unzipTarget.resolve("WEB-INF/classes/package"),
-            unzipTarget.resolve("WEB-INF/classes/package/test.properties"),
-            unzipTarget.resolve("WEB-INF/lib"),
-            unzipTarget.resolve("WEB-INF/web.xml"));
-    assertThat(layers.classesLayer.getEntries())
-        .comparingElementsUsing(SOURCE_FILE_OF)
-        .containsExactly(
-            unzipTarget.resolve("WEB-INF/classes/HelloWorld.class"),
-            unzipTarget.resolve("WEB-INF/classes/empty_dir"),
-            unzipTarget.resolve("WEB-INF/classes/package"),
-            unzipTarget.resolve("WEB-INF/classes/package/Other.class"));
-  }
-
   @Test
   public void testCreateContainerBuilder_noErrorIfWebInfClassesDoesNotExist()
       throws IOException, InvalidImageReferenceException, CacheDirectoryCreationException {
     temporaryFolder.newFolder("WEB-INF", "lib");
     setUpWarProject(temporaryFolder.getRoot().toPath());
-    setupBuildContext("/anything"); // should pass
+    setupBuildContext(); // should pass
```

Comment:
Ah sorry, missed this earlier. The behavior seems a little vague here. Adding a few tests on what setUpBuildContext results in might help?

Reply:
There are tests to check the layers. These tests are like follow-ups for successful tests. For example, this one ensures "noErrorIfWebInfClassesDoesNotExist." Or, do you have any ideas?

Reply:
Ah I see. Hm ideally, we want to make the test to be as explicit as possible and make it so that it's easier to understand the behavior of the method without us having to look through the code. In this test, for example,  I know that calling `setupBuildContext` doesn't throw an error but there isn't a way for me to tell if we are creating the correct layers (unless I take a look at the code itself)? It looks like that other tests use wars that have both `Web-INF/Classes` and `Web-INF/Lib` so in this test, maybe asserting on the layers would help in spotting the difference between `War with both Web-INF/Classes and Web-INF/Lib` and `War without Web-INF/Classes`. But let me know what you think.

Reply:
Here's my perspective: basically, a unit test should test one thing.

The other test verifies if we are creating the correct layers. The test data happens to have both `WEB-INF/classes` and `WEB-INF/lib`. If we want to test correct layer creation with a slightly different test data, I think we should create a new unit test for it (which may be meaningful to see the difference of the layer outcome as you said, but I think isn't really necessary given the coverage of the other test).

Now, this test is to ensure that calling `MavenProjectProperties.createJibContainerBuilder()` does not throw an exception if there are no `.class` files. (OTOH, for non-WAR, it does throw an exception.) I agree `setupBuildContext()` hides some stuff, but since we're using `setupBuildContext()` in all other test to reduce boiler plate for calling `createJibContainerBuilder()`, it's following the same pattern. If we add layer-checking code into this test, I think not only it's testing two things but also distracts the reader from noticing what this test really verifies. We could add a comment, but often that's unnecessary with the test name `testCreateContainerBuilder_noErrorIfWebInfClassesDoesNotExist()`.

Reply:
It just feels a little strange that we're just calling `setupBuildContext` in our test without having some sort of an assert block. I totally agree with you that test name does a good job of explaining what the test is supposed to do, but at the moment there doesn't seem to be a difference between the test method and the code it is testing? 

And to your point about a test, testing one thing -> You're absolutely right that a test method should only test one behavior. But for this test,  the behavior we want to focus on is: "what happens when Web-inf/classes is missing". When we assert on the classes layer, the idea that no error was thrown becomes apparent when we check the layer being created. We would solve the issue of under-asserting instead of testing a second behavior?

Reply:
Yeah, you do have a point. Now asserting that it's successfully returning a built `JibContainerBuilder`.